Source/Core/Common/MsgHandler.cpp

```diff
@@ -131,26 +120,36 @@ bool MsgAlert(bool yes_no, MsgType style, const char* format, ...)
   if (s_msg_handler != nullptr &&
       (s_alert_enabled || style == MsgType::Question || style == MsgType::Critical))
   {
-    return s_msg_handler(caption, buffer, yes_no, style);
+    return s_msg_handler(caption, text.data(), yes_no, style);
   }
 
   return true;
 }
 
+// This is the first stop for gui alerts where the log is updated and the
+// correct window is shown, but only for legacy printf-style messages
+bool MsgAlert(bool yes_no, MsgType style, const char* format, ...)
+{
+  // Read message and write it to the log
+  const char* caption = GetCaption(style);
+  char buffer[2048];
+
+  va_list args;
+  va_start(args, format);
+  CharArrayFromFormatV(buffer, sizeof(buffer) - 1, s_str_translator(format).c_str(), args);
+  va_end(args);
+
+  return ShowMessageAlert(caption, buffer, yes_no, style);
+}
+
+// This is the first stop for gui alerts where the log is updated and the
+// correct window is shown, when using fmt
 bool MsgAlertFmtImpl(bool yes_no, MsgType style, fmt::string_view format,
                      const fmt::format_args& args)
 {
   const char* caption = GetCaption(style);
   const auto message = fmt::vformat(format, args);
-  ERROR_LOG_FMT(MASTER_LOG, "{}: {}", caption, message);
-
-  // Don't ignore questions, especially AskYesNo, PanicYesNo could be ignored
-  if (s_msg_handler != nullptr &&
-      (s_alert_enabled || style == MsgType::Question || style == MsgType::Critical))
-  {
-    return s_msg_handler(caption, message.c_str(), yes_no, style);
-  }
 
-  return true;
+  return ShowMessageAlert(caption, message.c_str(), yes_no, style);
```

Comment:
The use of `message.c_str()` means the previous `string_view` change doesn't do anything.  Will replace with just `message`.